### PR TITLE
[gym] hard error on xcode 8.3 with legacy api

### PR DESCRIPTION
**Pull request by hjanuschka**
hard error if xcode >= 8.3 is used and `use_legacy_build_api` is set.

error message modeled after this https://github.com/fastlane/fastlane/issues/8672#issuecomment-290544806

bildschirmfoto 2017-04-03 um 08 34 50

> originally i wanted to remove all code paths with `use_legacy_build_api` but maybe there are some folks still using xode8.2 or below.

### CODE REVIEW

**Comment by nafu** [comment]
These changes are not enough?
https://github.com/fastlane/fastlane/pull/8692 https://github.com/fastlane/fastlane/pull/8717

**Comment by hjanuschka** [comment]
@nafu  they only trigger on packaging errors, not if for example an ipa is uploaded and later on rejected on apple servers. - i think we should start (with 8.3) to hard disallow legacy api, to protect users from following errors, and therefore keeping issue impact low.

File: gym/lib/gym/options.rb [hunk 1]
```diff
@@ -9,6 +9,23 @@ def self.available_options
...
+      UI.important "and update the gym call to include the export method like this:"
+      UI.important "== App Store Builds =="
+      UI.error '     gym(scheme: "MyScheme", export_method: "app-store")'
+      UI.important "==  Ad Hoc Builds =="
+      UI.error '     gym(scheme: "MyScheme", export_method: "ad-hoc")'
+      UI.important "== Development Builds =="
+      UI.error '     gym(scheme: "MyScheme", export_method: "development")'
+      UI.important "== In-House Enterprise Builds =="
+      UI.error '     gym(scheme: "MyScheme", export_method: "enterprise")'
+      UI.important "If you run into a code signing error, please check out our troubleshooting guide and common issue and how to solve them."
```
Comment (KrauseFx) [inline gym/lib/gym/options.rb]: Should be

> If you run into a code signing error, please check out our troubleshooting guide for more information on how to solve the most common issues

**Review by KrauseFx — CHANGES_REQUESTED** [review]
This is a great change for sure, could you add a test for this too? Just that `use_legacy_build_api` causes a user error. Should be simple to add 👍

File: gym/lib/gym/options.rb (hunk 1 above)
Comment (hjanuschka) [inline gym/lib/gym/options.rb]: done

**Comment by hjanuschka** [comment]
@KrauseFx  spec's added

File: gym/lib/gym/options.rb [hunk 2]
```diff
@@ -104,6 +122,9 @@ def self.plain_options
                                        if value
                                          UI.important "Don't use this option any more, as it's deprecated by Apple"
                                        end
+                                       if Gym::Xcode.legacy_api_deprecated?
+                                         Gym::Options.legacy_api_note
```
Comment (KrauseFx) [inline gym/lib/gym/options.rb]: At fastlane very often we use `method_name!` with a `!` to clarify that this is either modifying the object, or raising an exception if something goes wrong, could you rename the method?

**Review by KrauseFx — APPROVED** [review]
Looks great, one small things, and feel free to merge. Thank you ❤️

**Comment by nafu** [comment]
Nice! 🚀

**Pull request merged**